src/app/main.rs

```diff
@@ -420,13 +421,22 @@ where
                     .push(buffer::layer(MAX_IN_FLIGHT))
                     .push(timeout::layer(config.bind_timeout))
                     .push(limit::layer(MAX_IN_FLIGHT))
+                    .push(strip_header::layer(super::DST_OVERRIDE_HEADER))
                     .push(router::layer(|req: &http::Request<_>| {
-                        let addr = super::http_request_authority_addr(req)
-                            .or_else(|_| super::http_request_host_addr(req))
-                            .or_else(|_| super::http_request_orig_dst_addr(req))
-                            .ok();
-                        debug!("outbound addr={:?}", addr);
-                        addr
+                        super::http_request_l5d_override_dst_addr(req)
+                            .ok()
+                            .map(|override_addr| {
+                                debug!("outbound addr={:?}; dst-override", override_addr);
+                                override_addr
+                            })
+                            .or_else(|| {
+                                let addr = super::http_request_authority_addr(req)
+                                    .or_else(|_| super::http_request_host_addr(req))
+                                    .or_else(|_| super::http_request_orig_dst_addr(req))
+                                    .ok();
```

Comment:
nit: I think you only have to have an `.ok()` once in this callchain (also one in L427)

Reply:
Yep I can change it. It will change how it get's printed in `debug!` if that is okay - `outbound addr=Ok(...)` instead of `outbound addr=Some(...)`